### PR TITLE
[main] Update Docker images, queues, etc.

### PR DESCRIPTION
**Pull request by dougbu**
- part of dotnet/aspnetcore-internal#3950
  - also touches on #36032
- move to Helix queues for Alpine 3.14, OSX 10.15, and (for Arm64) Debian 11 (not 9)
  - use OSX 11.00 when testing PRs and rolling builds; reduce 10.15 usage to scheduled runs
  - remove duplication between PRs / rolling builds and scheduled runs
- build source-index on `windows-latest` (not `vs2017-win2016`)
- update Helix Docker images for Debian.11.Amd64.Open and Fedora.34.Amd64.Open

nits:
- don't skip unused Helix queues
- remove versions from pipeline job display names
  - some were already outdated; rest will be confusing in the future
- remove comments about unused Helix queues

### CODE REVIEW

File: src/Testing/src/xunit/HelixConstants.cs [hunk 1]
```diff
@@ -6,6 +6,6 @@ namespace Microsoft.AspNetCore.Testing;
 public static class HelixConstants
 {
     public const string Windows10Arm64 = "Windows.10.Arm64v8.Open;";
-    public const string DebianArm64 = "Debian.9.Arm64.Open;";
+    public const string DebianArm64 = "Debian.11.Arm64.Open;";
```
Comment (dougbu) [inline src/Testing/src/xunit/HelixConstants.cs:9]: We skip some tests on Debian 9 Arm64 e.g. see #28090. May be worth testing if they work on Debian 11 Arm64.

File: .azure/pipelines/ci.yml [hunk 2]
```diff
@@ -739,7 +739,7 @@ stages:
     parameters:
       platform:
         name: 'Managed'
-        container: 'mcr.microsoft.com/dotnet-buildtools/prereqs:centos-7-f39df28-20191023143754'
+        container: 'mcr.microsoft.com/dotnet-buildtools/prereqs:centos-7-20210714125435-9b5bbc2'
```
Comment (dougbu) [inline .azure/pipelines/ci.yml:742]: Is this update valid❔ Should we instead move to `centos-7-source-build-20210714125450-5d87b80`❔

**Comment by dougbu** [comment]
``` diff
--- ".\\helix.matrix.before"    2021-11-16 13:38:59.593746300 -0800
+++ ".\\helix.matrix.after"     2021-11-16 13:21:31.644538900 -0800
@@ -1,14 +1,11 @@
-(Alpine.312.Amd64.Open)Ubuntu.1804.Amd64.Open@mcr.microsoft.com/dotnet-buildtools/prereqs:alpine-3.12-helix-20200908125345-56c6673
-(Debian.11.Amd64.Open)ubuntu.1804.amd64.open@mcr.microsoft.com/dotnet-buildtools/prereqs:debian-11-helix-amd64-20210304164428-5a7c380
-(Debian.9.Arm64.Open)Ubuntu.1804.Armarch.Open@mcr.microsoft.com/dotnet-buildtools/prereqs:debian-9-helix-arm64v8-a12566d-20190807161036
-(Fedora.34.Amd64.Open)ubuntu.1804.amd64.open@mcr.microsoft.com/dotnet-buildtools/prereqs:fedora-34-helix-20210728124700-4f64125
+(Alpine.314.Amd64.Open)Ubuntu.1804.Amd64.Open@mcr.microsoft.com/dotnet-buildtools/prereqs:alpine-3.14-helix-amd64-20210910135833-1848e19
+(Debian.11.Amd64.Open)Ubuntu.1804.Amd64.Open@mcr.microsoft.com/dotnet-buildtools/prereqs:debian-11-helix-amd64-20211001171307-0ece9b3
+(Debian.11.Arm64.Open)Ubuntu.1804.Armarch.Open@mcr.microsoft.com/dotnet-buildtools/prereqs:debian-11-helix-arm64v8-20211001171229-97d8652
+(Fedora.34.Amd64.Open)Ubuntu.1804.Amd64.Open@mcr.microsoft.com/dotnet-buildtools/prereqs:fedora-34-helix-20210924174119-4f64125
 (Mariner)Ubuntu.1804.Amd64.Open@mcr.microsoft.com/dotnet-buildtools/prereqs:cbl-mariner-1.0-helix-20210528192219-92bf620
-OSX.1014.Amd64.Open
-OSX.1100.Amd64.Open
+OSX.1015.Amd64.Open
 Redhat.7.Amd64.Open
-Ubuntu.1804.Amd64.Open
 Ubuntu.2004.Amd64.Open
 Windows.10.Amd64.Server20H2.Open
 Windows.10.Arm64v8.Open
-Windows.11.Amd64.ClientPre.Open
 Windows.Amd64.Server2022.Open
```

**Comment by dougbu** [comment]
``` diff
--- ".\\quarantined.pr.before"  2021-11-16 13:57:28.157821100 -0800
+++ ".\\quarantined.pr.after"   2021-11-16 13:53:39.075019400 -0800
@@ -1,4 +1,4 @@
-(Fedora.34.Amd64.Open)ubuntu.1804.amd64.open@mcr.microsoft.com/dotnet-buildtools/prereqs:fedora-34-helix-20210728124700-4f64125
-OSX.1014.Amd64.Open
+(Fedora.34.Amd64.Open)Ubuntu.1804.Amd64.Open@mcr.microsoft.com/dotnet-buildtools/prereqs:fedora-34-helix-20210924174119-4f64125
+OSX.1100.Amd64.Open
 Ubuntu.1804.Amd64.Open
 Windows.11.Amd64.ClientPre.Open
```

`aspnetcore-ci` Helix job runs on the same queues before and after except it does no Fedora 34 testing.

File: .azure/pipelines/benchmarks.yml [hunk 3]
```diff
@@ -10,7 +10,7 @@ jobs:
 - template: jobs/default-build.yml
   parameters:
     jobName: Windows_Build
-    jobDisplayName: "Build only : Windows"
+    jobDisplayName: "Build: Windows"
```
[the remark this answers is not in the document]
Comment (dougbu) [reply .azure/pipelines/benchmarks.yml:13]: @sebastienros does anything depend on this job name❔ Is the YAML even used❔

File: .azure/pipelines/ci.yml (hunk 2 above)
Comment (dougbu) [reply .azure/pipelines/ci.yml:742]: @MichaelSimons @mthalman @crummel❔

File: src/Testing/src/xunit/HelixConstants.cs (hunk 1 above)
Comment (dougbu) [reply src/Testing/src/xunit/HelixConstants.cs:9]: @Tratcher any chance #28090 can be avoided on Debian 11 Arm64❔

File: eng/targets/Helix.Common.props [hunk 4]
```diff
@@ -22,9 +22,9 @@
...
     <HelixAvailableTargetQueue Include="Ubuntu.1804.Amd64.Open" Platform="Linux" />
     <HelixAvailableTargetQueue Include="Windows.11.Amd64.ClientPre.Open" Platform="Windows" />
-    <HelixAvailableTargetQueue Include="OSX.1014.Amd64.Open" Platform="OSX" />
+    <HelixAvailableTargetQueue Include="OSX.1100.Amd64.Open" Platform="OSX" />
   </ItemGroup>
-  
+
   <!-- x64 Quarantined-only (quarantined-pr.yml and quarantined-tests.yml) test queues -->
   <ItemGroup Condition="'$(IsRequiredCheck)' == 'true' AND '$(TargetArchitecture)' == 'x64' AND '$(_UseHelixOpenQueues)' == 'true' AND '$(RunQuarantinedTests)' == 'true'">
     <HelixAvailableTargetQueue Include="$(HelixQueueFedora34)" Platform="Linux" />
```
[the remark this answers is not in the document]
Comment (wtgodbe) [reply eng/targets/Helix.Common.props:30]: I think this was the queue w/ `libmsquic` installed, we should keep it in both (at least for now) since it's the only place where we can run HTTP/3 tests on Linux (CC @JamesNK @Tratcher @adityamandaleeka)

File: eng/targets/Helix.Common.props [hunk 5]
```diff
@@ -33,25 +33,22 @@
...
-    <HelixAvailableTargetQueue Include="$(HelixQueueAlpine312)" Platform="Linux" />
-    <HelixAvailableTargetQueue Include="$(HelixQueueMariner)" Platform="Linux" />    
+    <HelixAvailableTargetQueue Include="$(HelixQueueMariner)" Platform="Linux" />
   </ItemGroup>
 
   <!-- x64 Queues for public helix-matrix.yml and quarantine pipelines, Windows cases-->
   <ItemGroup Condition="'$(TargetArchitecture)' == 'x64' AND '$(IsHelixDaily)' == 'true' AND '$(_UseHelixOpenQueues)' == 'true'">
     <HelixAvailableTargetQueue Include="Windows.10.Amd64.Server20H2.Open" Platform="Windows" />
-    <HelixAvailableTargetQueue Include="Windows.11.Amd64.ClientPre.Open" Platform="Windows" />
```
Comment (wtgodbe) [inline eng/targets/Helix.Common.props:54]: Why remove this?

File: eng/targets/Helix.targets [hunk 6]
```diff
@@ -10,14 +10,12 @@
 
   <PropertyGroup Condition="'$(TestDependsOnPlaywright)' == 'true'">
     <SkipHelixQueues>
-      $(HelixQueueAlpine312);
+      $(HelixQueueAlpine314);
```
Comment (dougbu) [inline eng/targets/Helix.targets:13]: Worth removing this line and running `aspnetcore-helix-matrix` again❔ It's unclear how to determine whether Playwright is working everywhere given all or most Playwright tests are quarantined and at least the `BlazorWasmTemplateTest` tests mostly fail in any case. @javiercn ❔

**Review by dougbu — COMMENTED** [review]
Notes
- `aspnetcore-helix-matrix` failures are about flakiness that exists in our regular runs
- `aspnetcore-quarantined-pr` pipeline times out for 'main' builds fairly frequently

File: eng/targets/Helix.Common.props (hunk 4 above)
[the remark this answers is not in the document]
Comment (dougbu) [reply eng/targets/Helix.Common.props:30]: Do you mean all HTTP/3 tests are quarantined❔ My question was only about the duplication between `aspnetcore-quarantined-tests` (a bit lower down) and `aspnetcore-quarantined-pr`.

File: eng/targets/Helix.Common.props (hunk 5 above)
Comment (dougbu) [reply eng/targets/Helix.Common.props:54]: &hellip; because the line was part of the 💯% overlap of the queues used in `aspnetcore-ci` with what we use in `aspnetcore-helix-matrix` and `aspnetcore-quarantined-tests`. I also removed Ubuntu.1804.Amd64.Open and OSX.1100.Amd64.Open from the scheduled runs for the same reason.

See https://github.com/dotnet/aspnetcore/pull/38427/files#diff-fecf6f98b5c2a5cc5c66f045a93d30421b7fedcbdd4cacefc2605cd61a9b89afR23-R25

File: eng/targets/Helix.Common.props (hunk 4 above)
[the remark this answers is not in the document]
Comment (wtgodbe) [reply eng/targets/Helix.Common.props:30]: No, most are no longer quarantined. As long as the Fedora34 queue is used in at least one non-quarantine job, it should be fine to remove it from either `quar-tests` or `quar-pr`

[the remark this answers is not in the document]
Comment (dougbu) [reply eng/targets/Helix.Common.props:30]: :ok: I'll remove this block in the next iteration. That'll realign `aspnetcore-ci` and `aspnetcore-quarantined-pr` queues. `aspnetcore-helix-matrix` and `aspnetcore-quarantined-tests` were already aligned and _both_ include the Fedora 34 queue.

File: eng/targets/Helix.targets (hunk 6 above)
Comment (dougbu) [reply eng/targets/Helix.targets:13]: @wtgodbe you mentioned `Alpine 3.14 image doesn't work with Playwright` in #36032. How did you confirm that❔

Comment (HaoK) [reply eng/targets/Helix.targets:13]: We can basically safely ignore playwright now, I believe the only place that's running now is in the components pipleine, @TanayParikh can you confirm that's true?

Comment (wtgodbe) [reply eng/targets/Helix.targets:13]: Here are the original PRs where I:

Moved to 3.14: https://github.com/dotnet/aspnetcore/pull/36845
Updated to a 3.14 queue w/ Python: https://github.com/dotnet/aspnetcore/pull/36929
- Associated Helix-Matrix: https://dev.azure.com/dnceng/public/_build/results?buildId=1384007&view=results
Reverted back to 3.12 after persistent failures: https://github.com/dotnet/aspnetcore/pull/36951

The linked helix-matrix might have info that isn't in the PRs - all I remember is that the failure I mentioned in the issue was persistent. I couldn't find anything more in emails.

File: src/ProjectTemplates/test/GrpcTemplateTest.cs [hunk 7]
```diff
@@ -35,7 +35,7 @@ public ITestOutputHelper Output
     }
 
     [ConditionalFact]
-    [SkipOnHelix("Not supported queues", Queues = "Windows.7.Amd64;Windows.7.Amd64.Open;Windows.81.Amd64.Open;All.OSX;" + HelixConstants.Windows10Arm64 + HelixConstants.DebianArm64)]
+    [SkipOnHelix("Not supported queues", Queues = "All.OSX;" + HelixConstants.Windows10Arm64 + HelixConstants.DebianArm64)]
```
Comment (HaoK) [inline src/ProjectTemplates/test/GrpcTemplateTest.cs:38]: While we are touching this, we should probably do SkipHelixArm in the csproj instead of the more fragile skipping of the current 2 arm queues. `<SkipHelixArm>true</SkipHelixArm>`

**Review by HaoK — APPROVED** [review]

File: eng/targets/Helix.targets (hunk 6 above)
Comment (wtgodbe) [reply eng/targets/Helix.targets:13]: Maybe @HaoK remembers something I don't

**Review by wtgodbe — APPROVED** [review]

File: src/ProjectTemplates/test/GrpcTemplateTest.cs (hunk 7 above)
Comment (dougbu) [reply src/ProjectTemplates/test/GrpcTemplateTest.cs:38]: This is a single test `class` among many in ProjectTemplates.Tests and we don't have a `[SkipHelixArm]` attribute. Could however add another special case in `SkipOnHelixAttribute` e.g. `All.Arm64` that's decided using `Contains("arm64")`. Is that your preference❔

In any case, what happens if / when #28090 is fixed on Debian 11 but not Windows 10, or visa versa❔ I was going to fork this branch to narrow the "full matrix" to just Debian 11 Arm64 and remove the queue from `[SkipOnHelix]`s soon.

File: eng/targets/Helix.targets (hunk 6 above)
Comment (dougbu) [reply eng/targets/Helix.targets:13]: >  believe the only place that's running now is in the components pipeline.

The Components pipeline mainly runs Microsoft.AspNetCore.Components.E2ETests and those rely on Selenium, not Playwright. In addition, those tests don't run on Helix.

On the other hand, Playwright is used only in BlazorTemplates.Tests. Those tests are all quarantined and never run on any Linux e.g. from a recent `aspnetcore-quarantined-tests` run of 'main':

image

In general, can that restriction be removed or reduced (adding a Linux platform)❔

File: .azure/pipelines/benchmarks.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (dougbu) [reply .azure/pipelines/benchmarks.yml:13]: /ping @sebastienros

File: .azure/pipelines/ci.yml (hunk 2 above)
Comment (dougbu) [reply .azure/pipelines/ci.yml:742]: Thanks @MichaelSimons❕ I'll leave this as-is 😺

File: eng/targets/Helix.targets (hunk 6 above)
Comment (dougbu) [reply eng/targets/Helix.targets:13]: > In general, can that restriction be removed or reduced (adding a Linux platform)❔

@javiercn and @dotnet/aspnet-blazor-eng (because you're likely the most familiar w/ the `$(TestDependsOnPlaywright)` exclusions) please chime in here. I see Playwright is fully supported on Ubuntu 18.04 and 20.04 but we're running our Playwright tests on neither. I haven't searched enough to understand why Ubuntu 18.04 is also not running Playwright tests.

File: src/ProjectTemplates/test/GrpcTemplateTest.cs (hunk 7 above)
Comment (dougbu) [reply src/ProjectTemplates/test/GrpcTemplateTest.cs:38]: /ping @HaoK

File: src/Testing/src/xunit/HelixConstants.cs (hunk 1 above)
Comment (dougbu) [reply src/Testing/src/xunit/HelixConstants.cs:9]: /ping @Tratcher

**Review by dougbu — COMMENTED** [review]
Generally looking to resolve questions before merging this PR&hellip;

File: .azure/pipelines/benchmarks.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (wtgodbe) [reply .azure/pipelines/benchmarks.yml:13]: I think `dependsOn` is based on `jobName`, not `jobDisplayName`, so it should be fine to change this

File: eng/targets/Helix.targets (hunk 6 above)
Comment (wtgodbe) [reply eng/targets/Helix.targets:13]: @javiercn @dotnet/aspnet-blazor-eng any thoughts on Doug's question above?

File: src/ProjectTemplates/test/GrpcTemplateTest.cs (hunk 7 above)
Comment (wtgodbe) [reply src/ProjectTemplates/test/GrpcTemplateTest.cs:38]: Any objections @HaoK?

File: src/Testing/src/xunit/HelixConstants.cs (hunk 1 above)
Comment (wtgodbe) [reply src/Testing/src/xunit/HelixConstants.cs:9]: I'll mention this in the issue in case we want to un-skip those in a separate PR

**Comment by wtgodbe** [comment]
We have one test failure on OSX 10.15:

> [FAIL]
[xUnit.net 00:01:41.07]       System.Threading.Tasks.TaskCanceledException : The request was canceled due to the configured HttpClient.Timeout of 100 seconds elapsing.
[xUnit.net 00:01:41.07]       ---- System.TimeoutException : A task was canceled.
[xUnit.net 00:01:41.07]       -------- System.Threading.Tasks.TaskCanceledException : A task was canceled.
[xUnit.net 00:01:41.07]       Stack Trace:
[xUnit.net 00:01:41.07]            at System.Net.Http.HttpClient.HandleFailure(Exception e, Boolean telemetryStarted, HttpResponseMessage response, CancellationTokenSource cts, CancellationToken cancellationToken, CancellationTokenSource pendingRequestsCts)
[xUnit.net 00:01:41.07]            at System.Net.Http.HttpClient.GetStringAsyncCore(HttpRequestMessage request, CancellationToken cancellationToken)
[xUnit.net 00:01:41.07]         /_/src/Identity/test/Identity.Test/IdentityUIScriptsTest.cs(86,0): at Microsoft.AspNetCore.Identity.Test.IdentityUIScriptsTest.IdentityUI_ScriptTags_FallbackSourceContent_Matches_CDNContent(ScriptTag scriptTag)
[xUnit.net 00:01:41.07]         --- End of stack trace from previous location ---
[xUnit.net 00:01:41.07]         ----- Inner Stack Trace -----
[xUnit.net 00:01:41.07]         
[xUnit.net 00:01:41.07]         ----- Inner Stack Trace -----
[xUnit.net 00:01:41.07]         /_/src/Identity/test/Identity.Test/RetryHandler.cs(56,0): at Microsoft.AspNetCore.Identity.Test.RetryHandler.SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
[xUnit.net 00:01:41.07]            at System.Net.Http.HttpClient.GetStringAsyncCore(HttpRequestMessage request, CancellationToken cancellationToken)
[xUnit.net 00:01:41.07]       Output:
[xUnit.net 00:01:41.07]         Sending request 'GET - https://cdnjs.cloudflare.com/ajax/libs/jquery/3.5.1/jquery.min.js' 1 attempt.
[xUnit.net 00:01:41.07]         Request 'GET - https://cdnjs.cloudflare.com/ajax/libs/jquery/3.5.1/jquery.min.js' failed with System.Threading.Tasks.TaskCanceledException: A task was canceled.
[xUnit.net 00:01:41.07]            at System.Threading.Tasks.TaskCompletionSourceWithCancellation`1.WaitWithCancellationAsync(CancellationToken cancellationToken)
[xUnit.net 00:01:41.07]            at System.Net.Http.HttpConnectionPool.GetHttp11ConnectionAsync(HttpRequestMessage request, Boolean async, CancellationToken cancellationToken)
[xUnit.net 00:01:41.07]            at System.Net.Http.HttpConnectionPool.SendWithVersionDetectionAndRetryAsync(HttpRequestMessage request, Boolean async, Boolean doRequestAuth, CancellationToken cancellationToken)
[xUnit.net 00:01:41.07]            at System.Net.Http.RedirectHandler.SendAsync(HttpRequestMessage request, Boolean async, CancellationToken cancellationToken)
[xUnit.net 00:01:41.07]            at Microsoft.AspNetCore.Identity.Test.RetryHandler.SendAsync(HttpRequestMessage request, CancellationToken cancellationToken) in /_/src/Identity/test/Identity.Test/RetryHandler.cs:line 40
  Failed Microsoft.AspNetCore.Identity.Test.IdentityUIScriptsTest.IdentityUI_ScriptTags_FallbackSourceContent_Matches_CDNContent(scriptTag: https://cdnjs.cloudflare.com/ajax/libs/jquery/3.5.1/jquery.min.js) [1 m 40 s]
  Error Message:
   System.Threading.Tasks.TaskCanceledException : The request was canceled due to the configured HttpClient.Timeout of 100 seconds elapsing.
---- System.TimeoutException : A task was canceled.
-------- System.Threading.Tasks.TaskCanceledException : A task was canceled.
  Stack Trace:
     at System.Net.Http.HttpClient.HandleFailure(Exception e, Boolean telemetryStarted, HttpResponseMessage response, CancellationTokenSource cts, CancellationToken cancellationToken, CancellationTokenSource pendingRequestsCts)
   at System.Net.Http.HttpClient.GetStringAsyncCore(HttpRequestMessage request, CancellationToken cancellationToken)
   at Microsoft.AspNetCore.Identity.Test.IdentityUIScriptsTest.IdentityUI_ScriptTags_FallbackSourceContent_Matches_CDNContent(ScriptTag scriptTag) in /_/src/Identity/test/Identity.Test/IdentityUIScriptsTest.cs:line 86
--- End of stack trace from previous location ---
----- Inner Stack Trace -----

----- Inner Stack Trace -----
   at Microsoft.AspNetCore.Identity.Test.RetryHandler.SendAsync(HttpRequestMessage request, CancellationToken cancellationToken) in /_/src/Identity/test/Identity.Test/RetryHandler.cs:line 56
   at System.Net.Http.HttpClient.GetStringAsyncCore(HttpRequestMessage request, CancellationToken cancellationToken)
  Standard Output Messages:
 Sending request 'GET - https://cdnjs.cloudflare.com/ajax/libs/jquery/3.5.1/jquery.min.js' 1 attempt.
 Request 'GET - https://cdnjs.cloudflare.com/ajax/libs/jquery/3.5.1/jquery.min.js' failed with System.Threading.Tasks.TaskCanceledException: A task was canceled.
    at System.Threading.Tasks.TaskCompletionSourceWithCancellation`1.WaitWithCancellationAsync(CancellationToken cancellationToken)
    at System.Net.Http.HttpConnectionPool.GetHttp11ConnectionAsync(HttpRequestMessage request, Boolean async, CancellationToken cancellationToken)
    at System.Net.Http.HttpConnectionPool.SendWithVersionDetectionAndRetryAsync(HttpRequestMessage request, Boolean async, Boolean doRequestAuth, CancellationToken cancellationToken)
    at System.Net.Http.RedirectHandler.SendAsync(HttpRequestMessage request, Boolean async, CancellationToken cancellationToken)
    at Microsoft.AspNetCore.Identity.Test.RetryHandler.SendAsync(HttpRequestMessage request, CancellationToken cancellationToken) in /_/src/Identity/test/Identity.Test/RetryHandler.cs:line 40

@HaoK have you seen this one before? Should we just quarantine it?

**Comment by Tratcher** [comment]
External networking issue? We wouldn't normally quarantine a test for that.

That said, why is a test hitting external resources?

File: .azure/pipelines/benchmarks.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (sebastienros) [reply .azure/pipelines/benchmarks.yml:13]: our benchmarks don't use it. As far as I am concerned we can remove it.

**Comment by dougbu** [comment]
> That said, why is a test hitting external resources?

The failure is in the `Microsoft.AspNetCore.Identity.Test--net7.0` work item and it's running fine everywhere but on OSX 10.15. I suspect it's just a glitch but agree it would be nice if the test weren't using JS downloads from anywhere other than the test web app. @HaoK❔

**Comment by dougbu** [comment]
Separately, @Tratcher could you answer my question at https://github.com/dotnet/aspnetcore/pull/38427/files#r749957765 please❔

File: src/Testing/src/xunit/HelixConstants.cs (hunk 1 above)
Comment (dougbu) [reply src/Testing/src/xunit/HelixConstants.cs:9]: Ah, you overlapped me pinging @Tratcher again 😃

Comment (Tratcher) [reply src/Testing/src/xunit/HelixConstants.cs:9]: I don't have an answer, I'm not sure what the issue is.

File: eng/targets/Helix.targets (hunk 6 above)
Comment (dougbu) [reply eng/targets/Helix.targets:13]: I'm now leaning toward closing everything but the duplicate Fedora testing unchanged and getting this in tonight or tomorrow. The experts can deal with our limited coverage for the Blazor template tests and Playwright system compatibility.

File: src/Testing/src/xunit/HelixConstants.cs (hunk 1 above)
Comment (dougbu) [reply src/Testing/src/xunit/HelixConstants.cs:9]: :ok: thanks. I'll leave the skips on Helix for #28090 alone in this PR.

File: src/ProjectTemplates/test/GrpcTemplateTest.cs (hunk 7 above)
Comment (HaoK) [reply src/ProjectTemplates/test/GrpcTemplateTest.cs:38]: Yeah that's fine, All.Arm64 would be better long term, but whatever is easiest is fine for this PR

**Comment by HaoK** [comment]
We need the external hit as we want to ensure the contents of identity UI match the cdn, so its literally comparing against the cdn contents

**Comment by HaoK** [comment]
For that specific test, we can also just configure helix retries since that has a known external network dependency: https://github.com/dotnet/aspnetcore/blob/main/eng/test-configuration.json#L10 we should add a glob line for all of the script ui tests: "Microsoft.AspNetCore.Identity.Test.IdentityUIScriptsTest.*"

**Comment by dougbu** [comment]
Note the _entire_ work item failed in this case even though it was only `IdentityUI_ScriptTags_FallbackSourceContent_Matches_CDNContent()` that fell over. The problem here is the `HttpClient` timeout of 100 seconds is too long to retry the expected 5 times. We could change src/Identity/test/Identity.Test/RetryHandler.cs to use a shorter timeout for the request. That would be much better than retrying the entire work item.

And, on @Tratcher's overall point, we would be in a much better place if the exponential retry in `RetryHandler` weren't the only thing we backed off or if we weren't trying to retry so many times.

Bottom line, the test fails almost never but the `RetryHandler` in that test class isn't well set up.

**Comment by HaoK** [comment]
We could just get rid of using retry handler and let the helix work item retry the entire workitem, it doesn't take long its already on the machine and all of the identity tests take a few seconds to run in its entirety.

**Comment by dougbu** [comment]
Does the Helix SDK retry support retry just the single `Microsoft.AspNetCore.Identity.Test--net7.0` work item on a single platform❔ If yes, 🆗 I guess but if the scope is more narrow than that e.g. if it submitted to every platform in the matrix or retried all of our tests on one platform, just no.

**Comment by HaoK** [comment]
Why does it matter the scope?  If it only fails on OSX, it doesn't really hurt specifying broader retry policy for that test class

**Comment by dougbu** [comment]
> Why does it matter the scope? If it only fails on OSX, it doesn't really hurt specifying broader retry policy for that test class

@HaoK and I chatted offline and the scope is a single Helix agent i.e. retries just retry one test class on that one platform.

More generally, we're in agreement to

1. Remove `RetryHandler` in the Identity tests and (maybe) wherever else we have similar complications w/ fewer benefits than the Helix retry feature
2. Enable the Helix retry feature for this particular test class
3. (Optionally -- if it works) Reduce the 100 second `HttpClient` timeout because that eats into our Helix timeouts too quickly. (If the Helix timeout restarts when the agent starts a retry, this point isn't worth touching.)

**Comment by dougbu** [comment]
I'm worry that macOS-10.15 Helix machines are significantly slower or have worse network connections than the macOS 10.14 machines we were using before. The most recent `aspnetcore-helix-matrix` run for this PR failed in exactly the same way as before, w/ `IdentityUI_ScriptTags_FallbackSourceContent_Matches_CDNContent(...)` timeouts after trying to download files 3 times ☹️. Thoughts @ilyas1974 @MattGal❔

Separately from that and somewhat strangely, we aren't seeing timeouts of the `IdentityUI_ScriptTags_SubresourceIntegrityCheck(...)` tests and those should be using the same CDN links. May eventually need to download the files in a `static` method and save the files for use in these two tests. That's for later though.

**Comment by dougbu** [comment]
To clarify, the move that seems to be hurting our `IdentityUIScriptsTests` class is OSX.1014.Amd64.Open ➡️ OSX.1015.Amd64.Open.

**Comment by HaoK** [comment]
We could just choose to skip these identity tests on the OSX queue, we don't really need the OS coverage for this, there's no variation between OS for this feature area

**Comment by dougbu** [comment]
> We could just choose to skip these identity tests on the OSX queue, we don't really need the OS coverage for this, there's no variation between OS for this feature area

WFM though this remains a general problem. @wtgodbe is hitting similar problems w/ tests on macOS 10.15 e.g. in #38536. I'll add the `[SkipOnHelix]` attribute for the single queue (plus it's non-`.Open` alternative) in this PR but won't bother doing another `aspnetcore-helix-matrix` run.

Suggest we move to macOS-11 and OSX.1100.Amd64[.Open] in all of your PRs for dotnet/aspnetcore-internal#3950 except where we have `aspnetcore-helix-matrix` or similar @wtgodbe.

File: src/Identity/test/Identity.Test/IdentityUIScriptsTest.cs [hunk 8]
```diff
@@ -17,6 +17,7 @@
 
 namespace Microsoft.AspNetCore.Identity.Test;
 
+[SkipOnHelix("https://github.com/dotnet/aspnetcore/issues/38542", Queues="OSX.1015.Amd64.Open;OSX.1015.Amd64")] //slow
```
Comment (dougbu) [inline src/Identity/test/Identity.Test/IdentityUIScriptsTest.cs:20]: ☹️

Note I'm skipping the test class in case the failure would just move to `IdentityUI_ScriptTags_SubresourceIntegrityCheck(...)` if I skipped only `IdentityUI_ScriptTags_FallbackSourceContent_Matches_CDNContent(...)`. I double-checked and don't see evidence `IdentityUI_ScriptTags_SubresourceIntegrityCheck(...)` ran before (and of course not after) the timeouts of `IdentityUI_ScriptTags_FallbackSourceContent_Matches_CDNContent(...)`.

**Comment by dougbu** [comment]
> Separately from that and somewhat strangely, we aren't seeing timeouts of the `IdentityUI_ScriptTags_SubresourceIntegrityCheck(...)` tests and those should be using the same CDN links. May eventually need to download the files in a `static` method and save the files for use in these two tests. That's for later though.

This wasn't correct. `IdentityUI_ScriptTags_FallbackSourceContent_Matches_CDNContent(...)` tests didn't run in the failing builds AFAICT.

File: src/Identity/test/Identity.Test/IdentityUIScriptsTest.cs (hunk 8 above)
Comment (HaoK) [reply src/Identity/test/Identity.Test/IdentityUIScriptsTest.cs:20]: Yeah seems fine, we could go so far as skip All.OSX if you wanted since we really only need to run this on any single queue to ensure we are including the right file versions

Comment (dougbu) [reply src/Identity/test/Identity.Test/IdentityUIScriptsTest.cs:20]: `<whining>I don't wanna do another iteration</whining>` 😭

Plus, we haven't seen similar issues in other queues. We could add a `[RunOnlyOnLatestUbuntu]` attribute at some point in the future 😄

**Comment by dougbu** [comment]
Hmm, shouldn't be surprised that backporting changes that touch C# files doesn't work (after #38076). Bit unhappy the .props and .targets files aren't aligned. ☹️

I'll deal w/ this @wtgodbe

**Comment by MattGal** [comment]
> I'm worry that macOS-10.15 Helix machines are significantly slower or have worse network connections than the macOS 10.14 machines we were using before. The most recent `aspnetcore-helix-matrix` run for this PR failed in exactly the same way as before, w/ `IdentityUI_ScriptTags_FallbackSourceContent_Matches_CDNContent(...)` timeouts after trying to download files 3 times ☹️. Thoughts @ilyas1974 @MattGal❔
> 
> Separately from that and somewhat strangely, we aren't seeing timeouts of the `IdentityUI_ScriptTags_SubresourceIntegrityCheck(...)` tests and those should be using the same CDN links. May eventually need to download the files in a `static` method and save the files for use in these two tests. That's for later though.

_Pasting reply from Teams:_

I think it's a bit of a stretch to extrapolate "machines are not as good" from some HttpClient timeouts.  These machines aren't in an Azure data center so literally everything they do (fetch helix work items from Azure Service Bus, send events to Azure Event Hub, download payloads from Azure Storage accounts, etc) involves communicating with and downloading successfully from external resources. 

There is some variance in hardware to be found (some pools have a mix of minis and Pros;) but even the oldest, worst mac minis we have are using gigabit adapters built into their main boards connected to the same general network topology as the others.  If we see a pattern of a specific machine having this problem, we can certainly investigate but my suspicions lie with some kind of DoS prevention system with cloudflare.

When the vendors aren't using every port on the KVM I will fetch some hardware specs off a few machines but I don't think we can realistically blame compute power on a failure to download an 88 KB file from an external source in 100 seconds.
